spacy/tokens/doc.pyx

```diff
@@ -1743,6 +1743,124 @@ cdef class Doc:
                     j += 1
         return output
 
+    @cython.boundscheck(False)  # Deactivate bounds checking
```

Comment:
Did you test benchmark the overhead of bounds checks?

Reply:
No, I thought it was best practice to remove them once I'd established that all the tests passed and everything worked as expected?

Reply:
I don't think that it is enough. We have had code before that corrupts memory, but passed tests with flying colors 😅 .

Reply:
Bounds checks have no measurable impact on speed, so I've removed the annotations.